napari/resources/_icons.py

```diff
@@ -0,0 +1,53 @@
+import re
+from functools import lru_cache
+from pathlib import Path
+
+ICON_PATH = (Path(__file__).parent / 'icons').resolve()
+ICONS = {x.stem: x for x in ICON_PATH.iterdir()}
+
+
+def get_icon_path(name: str) -> str:
+    """Return path to an SVG in the theme icons."""
+    icon = ICONS.get(name)
+    if not icon:
+        raise ValueError(
+            f"unrecognized icon name: {name!r}. Known names: {ICONS}"
+        )
+    return str(icon)
+
+
+svg_elem = re.compile(r'(<svg[^>]*>)')
+svg_style = """<style type="text/css">
+path {{fill: {0}; opacity: {1};}}
+polygon {{fill: {0}; opacity: {1};}}
+circle {{fill: {0}; opacity: {1};}}
+rect {{fill: {0}; opacity: {1};}}
+</style>"""
```

Comment:
do you have any concern about quoting the colors ? my guess this might work with non hex color if we quote it, or use `{x!r}` I'm not too familiar with features supported by qt css/svg engine though.

Reply:
don't have any particular insight here, was just following the MDN svg style page, which omits quotes (even for non-hex colors): https://developer.mozilla.org/en-US/docs/Web/SVG/Element/style